### PR TITLE
Limit orders/receipt filled field

### PR DESCRIPTION
**Pull request by nenadV91**
# Summary

This PR fixes the issue number 3 mentioned by @elena-zh in this comment https://github.com/cowprotocol/cowswap/pull/1502#pullrequestreview-1194347899
Basically it uses the same logic as it works on Explorer for calculating and displaying the filled amounts

## To test
- go to limit-orders page
- create few different orders
- open the receipt modal
- open the explorer link in modal
- make sure the values are same in receipt as they are in explorer

### CODE REVIEW

**Comment by nenadV91** [comment]
I will fix that after I merge this PR and https://github.com/cowprotocol/cowswap/pull/1544 into https://github.com/cowprotocol/cowswap/pull/1502

**Review by alfetopito — COMMENTED** [review]
I suggest adding a comment point to the source of the code.
For when we update it here we also go and update on Explorer.

Best case it would come from the same place, but we are not there yet :)

**Review by elena-zh — APPROVED** [review]
Hey @nenadV91 , I have created #1548 issue for the last case I mentioned

File: src/cow-react/modules/limitOrders/pure/ReceiptModal/FilledField.tsx [hunk 1]
```diff
@@ -11,31 +13,116 @@ interface Props {
...
+    executedSellAmount,
+    executedFeeAmount,
+    filledAmount,
+  } = order
 
-  const filledPercentage = useMemo(() => {
-    if (!order || !order.filledPercentage) {
+  const touched = filledPercentage?.gt(0)
+
+  let mainToken
```
Comment (shoom3301) [inline src/cow-react/modules/limitOrders/pure/ReceiptModal/FilledField.tsx:33]: We always must set types to variables

File: src/cow-react/modules/limitOrders/pure/ReceiptModal/FilledField.tsx [hunk 2]
```diff
@@ -11,31 +13,116 @@ interface Props {
...
+  let mainAddress
+  let mainAmount
+  let swappedToken
+  let swappedAddress
+  let swappedAmount
+  let action
+
+  let filledAmountWithFee, swappedAmountWithFee
+  if (kind === OrderKind.SELL) {
```
Comment (shoom3301) [inline src/cow-react/modules/limitOrders/pure/ReceiptModal/FilledField.tsx:42]: Should be moved to a function

**Review by shoom3301 — APPROVED** [review]
Soft approve, but FilledField should be refactored

File: src/cow-react/modules/limitOrders/pure/ReceiptModal/FilledField.tsx (hunk 2 above)
Comment (nenadV91) [reply src/cow-react/modules/limitOrders/pure/ReceiptModal/FilledField.tsx:42]: Its based on this https://github.com/cowprotocol/explorer/blob/develop/src/components/orders/FilledProgress/index.tsx so I would keep it for now and then later we can refactor this in both places.

File: src/cow-react/modules/limitOrders/pure/ReceiptModal/FilledField.tsx [hunk 3]
```diff
@@ -11,31 +15,116 @@ interface Props {
...
-    if (!order || !order.filledPercentage) {
+  const touched = !!filledPercentage?.gt(0)
+
+  let mainToken: Token
+  let mainAddress: string
+  let mainAmount: CurrencyAmount<Token>
+  let swappedToken: Token
+  let swappedAddress: string
+  let swappedAmount: JSBI | undefined
+  let action: string
```
Comment (anxolin) [inline src/cow-react/modules/limitOrders/pure/ReceiptModal/FilledField.tsx:41]: what do @shoom3301 think of all this `let` declaration?

File: src/cow-react/modules/limitOrders/pure/ReceiptModal/FilledField.tsx [hunk 4]
```diff
@@ -11,31 +15,116 @@ interface Props {
...
+    mainAmount = buyAmount
+
+    swappedToken = inputToken
+    swappedAddress = sellToken
+    swappedAmount = executedSellAmount
+
+    // Buy orders need to add the fee, to the sellToken too (swappedAmount in this case)
+    filledAmountWithFee = filledAmount
+    swappedAmountWithFee = new BigNumber(swappedAmount?.toString() || '0').plus(executedFeeAmount || '0')
+  }
```
Comment (anxolin) [inline src/cow-react/modules/limitOrders/pure/ReceiptModal/FilledField.tsx:72]: it feels like all of the above could be moved to a function to keep the render function simpler and easier to reason about

**Pull request merged**